packages/babel-plugin-transform-nullish-coalescing-operator/test/fixtures/nullish-coalescing/transform-in-default/expected.js

```diff
@@ -0,0 +1,3 @@
+function foo(foo, bar = (_foo = foo, _foo !== null && _foo !== void 0 ? _foo : "bar")) {
+  var _foo;
```

Comment:
This is definitely incorrect, but we can't do better at the moment. It'll take a try-finally to get right, which is kinda sad.

```js
var _foo;

function foo(foo, bar = (_foo = foo, _foo !== null && _foo !== void 0 ? _foo : "bar")) {
  try {
    // function body
  } finally {
    _foo = void 0;
  }
}

Reply:
It's probably better to use an IIFE, no?

```js
function foo(foo, bar = function(){let _foo = foo; return _foo !== null && _foo !== void 0 ? _foo : "bar";}()) {}
```

Avoids leaking anything to the outer scope, if that's the concern.

Reply:
😄 Oh man, I totally overlooked that this value can be anything, including a IIFE.

Reply:
Kids, don't try this at home!